build.gradle

```diff
@@ -346,25 +350,25 @@ task tagRelease(type: Exec) {
 """
 }
 
-task javadoc(type: Javadoc) {
+tasks.register("javadoc", Javadoc) {
   title "Spock Framework API Documentation ($variantLessVersion)"
   destinationDir file("build/javadoc/$variantLessVersion")
-  source subprojects.javadoc.source
-  classpath = files(subprojects.javadoc.classpath)
+  source subprojects.collect { project -> project.tasks.named("javadoc").map { it.source } }
+  classpath = files(subprojects.collect { project -> project.tasks.named("javadoc").map { it.classpath } })
 }
 
-configureJavadoc(javadoc, true)
+configureJavadoc(tasks.named("javadoc"), true)
 
-task groovydoc(type: Groovydoc) {
+tasks.register("groovydoc", Groovydoc) {
   docTitle "Spock Framework API Documentation ($variantLessVersion)"
   windowTitle "Spock Framework API Documentation ($variantLessVersion)"
   destinationDir file("build/groovydoc/$variantLessVersion")
-  source subprojects.groovydoc.source
-  classpath = files(subprojects.groovydoc.classpath)
-  groovyClasspath = projects.spockCore.dependencyProject.groovydoc.groovyClasspath
+  source subprojects.collect { project -> project.tasks.named("groovydoc").map { it.source } }
+  classpath = files(subprojects.collect { project -> project.tasks.named("groovydoc").map { it.classpath } })
+  groovyClasspath = projects.spockCore.tasks.named("groovydoc").map { it.groovyClasspath }
```

Comment:
This does not work

Reply:
Fixed in 15998094c8041336c765c53fd0f1eef6f44dad10.

Reply:
No, still does not work

Reply:
c98ccca82d4fdb76b2cf7bc37eea016f3e3ef842

Reply:
It broken on master branch:

```
* What went wrong:
Some problems were found with the configuration of task ':groovydoc' (type 'Groovydoc').
  - Type 'org.gradle.api.tasks.javadoc.Groovydoc' property 'access' doesn't have a configured value.
    
    Reason: This property isn't marked as optional and no value has been configured.
    
    Possible solutions:
      1. Assign a value to 'access'.
      2. Mark property 'access' as optional.
    
    Please refer to https://docs.gradle.org/8.0.2/userguide/validation_problems.html#value_not_set for more details about this problem.
```

Reply:
Well, it still fails, but at least now it fails with the same error as `upstream/master` :-D